query_execution/PolicyEnforcerBase.hpp

```diff
@@ -163,6 +174,8 @@ class PolicyEnforcerBase {
   // Key = query ID, value = QueryManagerBase* for the key query.
   std::unordered_map<std::size_t, std::unique_ptr<QueryManagerBase>> admitted_queries_;
 
+  std::unordered_set<std::size_t> removed_query_ids_;
```

Comment:
Yes, and this set is used in [PolicyEnforcerBase::exitQuery](https://github.com/apache/incubator-quickstep/pull/111/files#diff-c12b9614fe3de39d47a95a80e1980f9cR114) to check `query_id` in [QueryInitiateResponseMessage](https://github.com/apache/incubator-quickstep/pull/111/files#diff-72cc64e73a6199a842b657ce03025a17R168).

It is required if we do above check, as some query (without any rebuilds) may finish earlier (thus being removed from `admitted_queries_`) than when the `QueryInitiateResponseMessage` replies back from a `Shiftboss`.

Finally, as a `TODO`, we could do some `GC` on `removed_query_ids_`, if we finish checks for older removed `query_id`s.